src/runtime/borrowed_account.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+
+const bincode = sig.bincode;
+
+const Epoch = sig.core.Epoch;
+const InstructionError = sig.core.instruction.InstructionError;
+const Pubkey = sig.core.Pubkey;
+
+const AccountSharedData = sig.runtime.AccountSharedData;
+const TransactionContext = sig.runtime.TransactionContext;
+const WLockGuard = sig.runtime.TransactionAccount.WLockGuard;
+
+const MAX_PERMITTED_ACCOUNTS_DATA_ALLOCATIONS_PER_TRANSACTION =
+    sig.runtime.program.system_program.MAX_PERMITTED_ACCOUNTS_DATA_ALLOCATIONS_PER_TRANSACTION;
```

Comment:
Nit: Maybe have the directory named as only `system`? ie drop the `_program` suffix, given everything is already within `system` directory.